src/main/java/org/hisp/quick/batchhandler/AbstractBatchHandler.java

```diff
@@ -246,6 +246,30 @@ public final void updateObject( T object )
         }
     }
 
+    /**
+     * Upserts the given object.
+     *
+     * @param object the object to upsert.
+     */
+    @Override
+    public final void upsertObject( T object )
+    {
+        final String sql = statementBuilder.getUpsertStatement(object);
+
+        log.debug("Upsert SQL: " + sql);
+
+        try
+        {
+            statement.executeUpdate(sql);
+        }
+        catch (SQLException ex)
+        {
+            log.error("Failed to upsert object", ex);
+            close();
+            throw new RuntimeException("Failed to upsert object", ex);

```

Comment:
Sonar dislikes using `RuntimeException` but I assume this is consistent with the other implementation methods?